controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -54,15 +56,34 @@ func (r *ClowdenvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	)
 
 	if ready, _ := VerifyClowdEnvReady(env); ready {
-		ns := env.Spec.TargetNamespace
-		r.Log.Info("Clowdenvironment ready", "ns-name", ns)
+		nsName := env.Spec.TargetNamespace
+		r.Log.Info("Clowdenvironment ready", "ns-name", nsName)
 
-		if err := CreateFrontendEnv(ctx, r.Client, ns, env); err != nil {
-			r.Log.Error(err, "Error encountered with frontend environment", "ns-name", ns)
-			UpdateAnnotations(ctx, r.Client, map[string]string{"env-status": "error"}, ns)
+		if err := CreateFrontendEnv(ctx, r.Client, nsName, env); err != nil {
+			r.Log.Error(err, "Error encountered with frontend environment", "ns-name", nsName)
+			UpdateAnnotations(ctx, r.Client, map[string]string{"env-status": "error", "status": "error"}, nsName)
 		} else {
-			r.Log.Info("Namespace ready", "ns-name", ns)
-			UpdateAnnotations(ctx, r.Client, map[string]string{"env-status": "ready"}, ns)
+			r.Log.Info("Namespace ready", "ns-name", nsName)
+			UpdateAnnotations(ctx, r.Client, map[string]string{"env-status": "ready", "status": "ready"}, nsName)
+
+			ns, err := GetNamespace(ctx, r.Client, nsName)
+			if err != nil {
+				r.Log.Error(err, "Could not retrieve newly created namespace", "ns-name", nsName)
+			}
+
+			if _, ok := ns.Annotations["completion-time"]; !ok {
+				nsCompletionTime := time.Now()
+				ns.Annotations["completion-time"] = nsCompletionTime.String()
+
```

Comment:
Whenever I can I advise developers not to re-type important strings like this all over their code. These strings you have like `completion-time` and `ready` and `completion-time` - they aren't really strings. They're symbols. And if you mistyped one somewhere, or if you needed to change them all at the same time you'd need to track them all down. I'd suggest throwing some consts at the top:

```golang
const (
    ENV_STATUS = "ready"
    COMPLETION_TIME = "completion-time"
    //etc 
)

//...

if _, ok := ns.Annotations[COMPLETION_TIME]; !ok {
				nsCompletionTime := time.Now()
				ns.Annotations[COMPLETION_TIME] = nsCompletionTime.String()
}
```

If the string is repeated, and it isn't intended for human consumption, this is the safest thing to do.